coral-common/src/main/java/com/linkedin/coral/common/transformers/SqlCallTransformer.java

```diff
@@ -5,37 +5,32 @@
  */
 package com.linkedin.coral.common.transformers;
 
-import java.util.ArrayList;
-import java.util.List;
-
 import com.google.common.collect.ImmutableList;
 
 import org.apache.calcite.rel.type.RelDataType;
 import org.apache.calcite.sql.SqlCall;
 import org.apache.calcite.sql.SqlIdentifier;
 import org.apache.calcite.sql.SqlNode;
-import org.apache.calcite.sql.SqlNodeList;
 import org.apache.calcite.sql.SqlOperator;
-import org.apache.calcite.sql.SqlSelect;
 import org.apache.calcite.sql.parser.SqlParserPos;
 import org.apache.calcite.sql.type.SqlReturnTypeInference;
 import org.apache.calcite.sql.validate.SqlUserDefinedFunction;
-import org.apache.calcite.sql.validate.SqlValidator;
+
+import com.linkedin.coral.common.utils.TypeDerivationUtil;
 
 
 /**
  * Abstract class for generic transformations on SqlCalls
  */
 public abstract class SqlCallTransformer {
-  private SqlValidator sqlValidator;
-  private final List<SqlSelect> topSelectNodes = new ArrayList<>();
+  private TypeDerivationUtil _typeDerivationUtil;
```

Comment:
nit: why the naming pattern staring with `_` is used here?

Reply:
thanks for pointing it out. Fixed it by removing `_`